### PR TITLE
feat: add Snowflake and Permissions types

### LINKED ISSUE #64: Snowflake type
we can make a Snowflake type like https://github.com/discordjs/discord.js/pull/5157

docs already say snowflake for snowflake properties

### PR DESCRIPTION
**Pull request by advaith1**
discordjs/discord.js#5157

closes #64

### CODE REVIEW

File: v8/payloads/channel.ts [hunk 1]
```diff
@@ -207,7 +209,7 @@ export interface APIMessage {
 	 *
 	 * See https://discord.com/developers/docs/topics/permissions#role-object
 	 */
-	mention_roles: string[];
+	mention_roles: Pick<APIRole, 'id'>[];
```
Comment (vladfrangu) [inline v8/payloads/channel.ts:212]: are these... objects? Because I think you meant `APIRole['id']`

File: v8/payloads/emoji.ts [hunk 2]
```diff
@@ -29,7 +31,7 @@ export interface APIEmoji extends APIPartialEmoji {
 	/**
 	 * Roles this emoji is whitelisted to
 	 */
-	roles?: string[];
+	roles?: Pick<APIRole, 'id'>[];
```
Comment (vladfrangu) [inline v8/payloads/emoji.ts:34]: ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: v8/payloads/channel.ts (hunk 1 above)
Comment (advaith1) [reply v8/payloads/channel.ts:212]: ah yeah oops

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Can you do permissions too?

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**